pkg/handler/eval_cache.go

```diff
@@ -70,13 +92,37 @@ func (ec *EvalCache) GetByTags(tags []string) []*entity.Flag {
 			}
 		}
 	}
+	return results
+}
 
-	values := make([]*entity.Flag, 0, len(results))
-	for _, f := range results {
-		values = append(values, f)
+func (ec *EvalCache) getByTagsALL(tags []string) map[uint]*entity.Flag {
+	results := map[uint]*entity.Flag{}
+	cache := ec.cache.Load().(*cacheContainer)
+
+	for i, t := range tags {
+		fSet, ok := cache.tagCache[t]
+		if !ok {
+			// no flags
+			return map[uint]*entity.Flag{}
+		}
+
+		if i == 0 {
+			// store all the flags
+			for fID, f := range fSet {
+				results[fID] = f
+			}
+		} else {
+			if len(results) > 0 {
```

Comment:
one more thing to optimize is that you can return early when `len(results) == 0`

Reply:
Done, after each delete loop we check if there is still any match to continue or return early.